### ISSUE TITLE
API Server not available.

### ISSUE THREAD

**Issue by pooleja** [Blockstream/esplora#308]
It looks like the bitcoin node has sync'd and the services are starting to come up, but the API server is either restarting or hung.  I see the same sort of message printed over and over.

```
2-Config { log: StdErrLog { verbosity: Info, quiet: false, show_level: true, timestamp: Millisecond, modules: [], writer: "stderr", color_choice: Never }, network_type: Bitcoin, db_path: "/data/electrs_bitcoin_db/mainnet/mainnet", daemon_dir: "/data/bitcoin", blocks_dir: "/data/bitcoin/blocks", daemon_rpc_addr: 127.0.0.1:8332, cookie: None, electrum_rpc_addr: 0.0.0.0:50001, http_addr: 127.0.0.1:3000, http_socket_file: Some("/var/electrs-rest.sock"), monitoring_addr: 0.0.0.0:4224, jsonrpc_import: false, light_mode: false, address_search: true, index_unspendables: false, cors: None, precache_scripts: Some("/srv/explorer/popular-scripts.txt"), utxos_limit: 500, electrum_txs_limit: 100000, electrum_banner: "Welcome to electrs-esplora 0.4.1", electrum_public_hosts: None, electrum_announce: false, tor_proxy: Some(127.0.0.1:9050) }
2-2021-04-22T18:28:01.920+00:00 - INFO - NetworkInfo { version: 200100, subversion: "/Satoshi:0.20.1/", relayfee: 0.00001 }
2-2021-04-22T18:28:03.600+00:00 - INFO - BlockchainInfo { chain: "main", blocks: 680169, headers: 680169, bestblockhash: "00000000000000000000f5e90a2b27239479b9b1a76c59020611f70e788fe314", pruned: false, verificationprogress: 0.9999978, initialblockdownload: Some(false) }
4-2021-04-22T18:28:27Z UpdateTip: new best=00000000000000000009c20c6419f18b52e5dce1c3b5522f1691268a16234dfb height=680170 version=0x3fffe000 log2_work=92.826171 tx=636051155 date='2021-04-22T18:27:56Z' progress=1.000000 cache=377.8MiB(2854361txo) warning='71 of last 100 blocks have unexpected version'
2-2021-04-22T18:28:34.569+00:00 - INFO - hash=00000000000000000000f5e90a2b27239479b9b1a76c59020611f70e788fe314 height=680169 @ 2021-04-22T18:21:02+00:00 (680170 left to index)
2-Config { log: StdErrLog { verbosity: Info, quiet: false, show_level: true, timestamp: Millisecond, modules: [], writer: "stderr", color_choice: Never }, network_type: Bitcoin, db_path: "/data/electrs_bitcoin_db/mainnet/mainnet", daemon_dir: "/data/bitcoin", blocks_dir: "/data/bitcoin/blocks", daemon_rpc_addr: 127.0.0.1:8332, cookie: None, electrum_rpc_addr: 0.0.0.0:50001, http_addr: 127.0.0.1:3000, http_socket_file: Some("/var/electrs-rest.sock"), monitoring_addr: 0.0.0.0:4224, jsonrpc_import: false, light_mode: false, address_search: true, index_unspendables: false, cors: None, precache_scripts: Some("/srv/explorer/popular-scripts.txt"), utxos_limit: 500, electrum_txs_limit: 100000, electrum_banner: "Welcome to electrs-esplora 0.4.1", electrum_public_hosts: None, electrum_announce: false, tor_proxy: Some(127.0.0.1:9050) }
2-2021-04-22T18:28:46.358+00:00 - INFO - NetworkInfo { version: 200100, subversion: "/Satoshi:0.20.1/", relayfee: 0.00001 }
2-2021-04-22T18:28:48.414+00:00 - INFO - BlockchainInfo { chain: "main", blocks: 680170, headers: 680170, bestblockhash: "00000000000000000009c20c6419f18b52e5dce1c3b5522f1691268a16234dfb", pruned: false, verificationprogress: 0.9999997, initialblockdownload: Some(false) }
```

Wondering if there is any way to troubleshoot this?

**Comment by pooleja** [comment]
I see `verificationprogress: 0.99999875` never quite moves to 100%.

**Comment by pooleja** [comment]
If anyone else runs into this... The system requirements that electrs lists as "not much more than a standard node" is fantasy land.  I ended up running out of disk space and even with a 4 CPU and 16 GB ram AWS EC2 instance it still took about a week to sync and for the API to become available using the blockstream published Docker container.  The max disk space when syncing I noticed was about 1.3 TB and the current block is 680834.  Currently after it is synced I see disk space usage at 1.112 TB.  I also updated IOPs on the machine's drive to 10k to try and match a SSD speed.

Takeaway... if the web page comes up but the APIs are not available then likely electrs is not synced.  Be prepared to wait.
